src/FindFunction.hs

```diff
@@ -0,0 +1,95 @@
+module FindFunction where
```

Comment:
Maybe we can move these into the polymorphism module since it's pretty small and these functions all seem to deal with finding the right function to use in a polymorphic context?

Reply:
Sure, I'll try that!